pkg/metrics/providers/graphite.go

```diff
@@ -21,27 +21,80 @@ import (
 	"encoding/json"
 	"fmt"
 	"io/ioutil"
+	"math"
 	"net/http"
 	"net/url"
 	"path"
 	"regexp"
+	"strconv"
 	"time"
 
 	flaggerv1 "github.com/fluxcd/flagger/pkg/apis/flagger/v1beta1"
 )
 
-type graphiteMetric struct {
-	Leaf          int         `json:"leaf"`
-	Context       interface{} `json:"context"`
-	Text          string      `json:"text"`
-	Expandable    int         `json:"expandable"`
-	ID            string      `json:"id"`
-	AllowChildren int         `json:"allowChildren"`
+type graphiteDataPoint struct {
+	Value     *float64
+	TimeStamp time.Time
 }
 
-type graphiteResponse []graphiteMetric
+func (gdp *graphiteDataPoint) UnmarshalJSON(data []byte) error {
 
-// GraphiteProvider executes Graphite queries.
+	var v []interface{}
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
```

Comment:
not sure if this is an issue or not - does golang runtime error if the array is not long enough? The following code expects eelments at [0] and [1]

Reply:
Golang will panic in that case. It is probably best to guard against that.